modules/light-clients/08-wasm/types/proposal_handle_test.go

```diff
@@ -78,202 +84,99 @@ func (suite *TypesTestSuite) TestCheckSubstituteAndUpdateStateGrandpa() {
-// 			suite.Require().NoError(err)
-// 			substituteTmClientState := substituteWasmClientStateData.(*ibctm.ClientState)
-
-// 			// update trusting period of substitute client state
-// 			substituteTmClientState.TrustingPeriod = time.Hour * 24 * 7
-
-// 			substituteTmClientStateBz, err := clienttypes.MarshalClientState(suite.chainA.App.AppCodec(), substituteTmClientState)
-// 			suite.Require().NoError(err)
-// 			substituteWasmClientState.Data = substituteTmClientStateBz
-// 			substitutePath.EndpointA.SetClientState(substituteWasmClientState)
-
-// 			// update substitute a few times
-// 			for i := 0; i < 3; i++ {
-// 				err := substitutePath.EndpointA.UpdateClient()
-// 				suite.Require().NoError(err)
-// 				// skip a block
-// 				suite.coordinator.CommitBlock(suite.chainA, suite.chainB)
-// 			}
-
-// 			// get updated substitute
-// 			substituteWasmClientState = suite.chainA.GetClientState(substitutePath.EndpointA.ClientID).(*types.ClientState)
-// 			err = suite.chainA.Codec.UnmarshalInterface(substituteWasmClientState.Data, &substituteWasmClientStateData)
-// 			suite.Require().NoError(err)
-// 			substituteTmClientState = substituteWasmClientStateData.(*ibctm.ClientState)
-
-// 			// test that subject gets updated chain-id
-// 			newChainID := "new-chain-id"
-// 			substituteTmClientState.ChainId = newChainID
+func (suite *TypesTestSuite) TestCheckSubstituteAndUpdateState() {
+	var substituteClientState exported.ClientState
+	contractErr := errors.New("contract error")
 
-// 			substituteTmClientStateBz, err = clienttypes.MarshalClientState(suite.chainA.App.AppCodec(), substituteTmClientState)
-// 			suite.Require().NoError(err)
-// 			substituteWasmClientState.Data = substituteTmClientStateBz
-// 			substitutePath.EndpointA.SetClientState(substituteWasmClientState)
-
-// 			subjectClientStore := suite.chainA.App.GetIBCKeeper().ClientKeeper.ClientStore(suite.chainA.GetContext(), subjectPath.EndpointA.ClientID)
-// 			substituteClientStore := suite.chainA.App.GetIBCKeeper().ClientKeeper.ClientStore(suite.chainA.GetContext(), substitutePath.EndpointA.ClientID)
-
-// 			expectedConsState := substitutePath.EndpointA.GetConsensusState(substituteWasmClientState.GetLatestHeight())
-// 			expectedProcessedTime, found := ibctm.GetProcessedTime(substituteClientStore, substituteWasmClientState.GetLatestHeight())
-// 			suite.Require().True(found)
-// 			expectedProcessedHeight, found := GetProcessedHeight(substituteClientStore, substituteWasmClientState.GetLatestHeight())
-// 			suite.Require().True(found)
-// 			expectedIterationKey := ibctm.GetIterationKey(substituteClientStore, substituteWasmClientState.GetLatestHeight())
+	testCases := []struct {
+		name     string
+		malleate func()
+		expErr   error
+	}{
+		{
+			"success",
+			func() {
+				suite.mockVM.RegisterSudoCallback(
+					types.CheckSubstituteAndUpdateStateMsg{},
```

Comment:
this will need to call `types.MigrateClientStoreMsg{}` instead, see [this pr](https://github.com/cosmos/ibc-go/pull/4974/files)